Added Multi-User and progress tracking functionality

harry6396 opened the pull request:
1. Added multi-user support.
2. Added feature to track the progress of each user.

onetruffle committed:
Added gamesClearedInTotal to the UserDefault database

onetruffle committed:
Added progress columns

onetruffle committed:
Granted mainapp resource permissions to the launcher

onetruffle committed:
Correction about sharedUserId

The attribute conflicts with another sharedUserId attribute in a different manifest file in the debug/release folder.

onetruffle committed:
Add columns

onetruffle committed:
Added coins to the table

onetruffle committed:
Added progress variable to each user

onetruffle committed:
Revert "Correction about sharedUserId"

This reverts commit fd5926b2d2c1ebd4ddb64f137704137b8c9c4b28.

onetruffle committed:
Revert "Granted mainapp resource permissions to the launcher"

This reverts commit 569ca0045065e3e98fdf869fd0becda667ae6625.

onetruffle committed:
Minor changes

Commit:
Changes

Commit:
Changes

Commit:
Changes

Commit:
Very minor changes

Very Minor changes

onetruffle committed:
Revert "Changes"

This reverts commit f2c112bd3534979f2a2a76d3619e2e468222d56a.

Commit:
Minor Change

onetruffle committed:
Fix sharedUserId for access

onetruffle committed:
Added a column

Commit:
Changes

Commit:
Minor changes

Commit:
Minor changes

Commit:
Minor changes

Commit:
Minor Changes

onetruffle committed:
Merge branch 'SanchitT/MultiUserFunctionality' into UserProfile_ProgressTracker_Integration

onetruffle committed:
.

onetruffle committed:
Merge branch 'SaitarunI/ProgressTracker' into UserProfile_ProgressTracker_Integration

onetruffle committed:
Update strings.xml

onetruffle committed:
Update User.java

Commit:
Minor changes

Commit:
minor change

Commit:
Update FirstScreen.java

Commit:
Changes

Commit:
Update build.gradle

Commit:
Update FirstScreen.java

onetruffle committed:
Update UserNameListDialog.java

onetruffle committed:
Merge branch 'UserProfile_ProgressTracker_Integration' of https://github.com/maqsoftware/GLEXP-Team-KitkitSchool-Hindi into UserProfile_ProgressTracker_Integration

Commit:
Revert "Update build.gradle"

This reverts commit 5ebff1b1a21e4fe81b4e35844d1b428041c435a2.

Commit:
Revert "Changes"

This reverts commit 3566ec09ed6ace6937cc5a3e94615b0b956113f6.

Commit:
Reverted the changes of gradle files

Commit:
Updated the file name and removed unused code

Commit:
Update MultiUserActivity.java

Commit:
Update MultiUserActivity.java

Commit:
Update MultiUserActivity.java

Commit:
Updated the UI of adding the user and removed unsued code

Commit:
Removed the unused variables

Commit:
Update MultiUserActivity.java

Commit:
Update in Manifest and multi_user layout

harry6396 commented on mainapp/proj.android-studio/app/release/AndroidManifest.xml:
```diff
@@ -1,7 +1,7 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.maq.xprize.kitkitschool.hindi"
-    android:sharedUserId="com.maq.xprize.kitkitschool.hindi"
+    android:sharedUserId="com.maq.kitkitschool"
```
Revert these changes

SanchitTaliyan replied:
Reverted

harry6396 commented on kitkitschoollogger/kitkitlogger/src/main/java/com/maq/kitkitProvider/KitkitDBHandler.java:
```diff
@@ -33,6 +33,12 @@
 
     public static final String COLUMN_ID = "_id";
     public static final String COLUMN_USERNAME = "username";
+   // public static final String COLUMN_MATHS = "maths";
```
Remove commented code

SanchitTaliyan replied:
Resolved

harry6396 commented on kitkitschoollogger/kitkitlogger/src/main/java/com/maq/kitkitProvider/KitkitDBHandler.java:
```diff
@@ -63,6 +69,12 @@
+            + COLUMN_GENDER + " TEXT,"
+            + COLUMN_AGE + " TEXT,"
+           // + COLUMN_ENGLISH + " INTEGER,"
+           // + COLUMN_MATHS + " INTEGER,"
```
Removed commented code

harry6396 commented on kitkitschoollogger/kitkitlogger/src/main/java/com/maq/kitkitProvider/KitkitDBHandler.java:
```diff
@@ -83,7 +95,13 @@
+            + COLUMN_AGE + " TEXT"
+           // + COLUMN_ENGLISH + " INTEGER,"
+          //  + COLUMN_MATHS + " INTEGER,"
+            //+ COLUMN_TOTAL + " INTEGER,"
```
Same as above

harry6396 commented on kitkitschoollogger/kitkitlogger/src/main/java/com/maq/kitkitProvider/KitkitDBHandler.java:
```diff
@@ -160,6 +178,12 @@ private void execRawQuery(SQLiteDatabase db, String[] arr_sql) {
+        //values.put(COLUMN_MATHS, user.getMaths());
+       // values.put(COLUMN_ENGLISH, user.getEnglish());
+       // values.put(COLUMN_TOTAL, user.getTotal());
+       // values.put(COLUMN_COINS, user.get_coins());
```
Same

SanchitTaliyan replied:
removed the redundant comments

harry6396 commented on kitkitschoollogger/kitkitlogger/src/main/java/com/maq/kitkitProvider/KitkitDBHandler.java:
```diff
@@ -180,6 +205,12 @@ public void addUser(User user) {
+               // COLUMN_MATHS,
+               // COLUMN_ENGLISH,
+               // COLUMN_TOTAL,
+              //  COLUMN_COINS,
```
Remove all commented code

SanchitTaliyan replied:
Removed the same

harry6396 commented on kitkitschoollogger/kitkitlogger/src/main/java/com/maq/kitkitProvider/User.java:
```diff
@@ -25,10 +27,16 @@
     private boolean _unlockFishBowl;
     private boolean _unlockWritingBoard;
     private boolean _finishWritingBoardTutorial;
+    /////////////////////
```
remove these lines

SanchitTaliyan replied:
removed

harry6396 commented on launcher/app/build.gradle:
```diff
@@ -12,15 +12,15 @@ android {
+
+    signingConfigs {
         debug {
             storeFile file('../debug.keystore')
```
Revert the changes

SanchitTaliyan replied:
Reverted

Commit:
Removed unused code

Commit:
Updated the shared user id

Commit:
Updated the shared user id of launcher release app

Commit:
Merge branch 'newmaster' into UserProfile_ProgressTracker_Integration

Commit:
Fixed the issue of coins

Commit:
Merge branch 'UserProfile_ProgressTracker_Integration' of https://github.com/maqsoftware/GLEXP-Team-KitkitSchool-Hindi into UserProfile_ProgressTracker_Integration

Commit:
Updated the SharedUserId, fixed the SetUser function, updated the launcher layout

Commit:
Changed the sharedUserId in release Android Manifest

onetruffle committed:
Resolved issues of severity "error" and "portability"

onetruffle committed:
Resolve issues of severity "performance"

onetruffle committed:
Resolve more issues of severity "performance"

onetruffle committed:
Resolve more issues of severity "performance"

Commit:
Changed the add and select buttons image in activity_multi_user

onetruffle committed:
fixes

onetruffle committed:
Resolve more issures of severity "performance"

Commit:
Changes in Add and Select user UI

Commit:
Merge branch 'newmaster' into UserProfile_ProgressTracker_Integration

onetruffle committed:
Resolved issues of severity "warning"

Commit:
Updated the Add and Select User layout

Commit:
Merge branch 'UserProfile_ProgressTracker_Integration' of https://github.com/maqsoftware/GLEXP-Team-KitkitSchool-Hindi into UserProfile_ProgressTracker_Integration

Commit:
Updated the layout of add user

Commit:
Fixed the layout for all devices

Commit:
Add User functionality and Select User UI progress

Commit:
Added Select User Functionality

Commit:
Updated Select User and Main Activity

In Select User added set current user and in Main Activity added the functionality to call Select User when clicked on user name.

Commit:
Updated Add User

After clicking picture picture will replace the button

onetruffle committed:
Resolve lint errors

onetruffle committed:
Severity "error"

Commit:
Updated the code to redirect the user to the multi-user activity

Commit:
Added code to fetch the current page

Commit:
Update MultiUserActivity.java

Commit:
Merge branch 'UserProfile_ProgressTracker_Integration' of https://github.com/maqsoftware/GLEXP-Team-KitkitSchool-Hindi into UserProfile_ProgressTracker_Integration

Commit:
Added Edit User fuctionality and cleaned the code

onetruffle committed:
Revert "Severity "error""

This reverts commit 3448658534fd3471230d69f7d00ec26255011c9f.

onetruffle committed:
Merge branch 'newmaster' into SaitarunI/StaticCodeAnalysis_6_11

ankitk20 commented on kitkitschoollogger/kitkitlogger/src/main/java/com/maq/kitkitProvider/KitkitDBHandler.java:
```diff
@@ -387,15 +406,18 @@ public String getCurrentUsername() {
 
         String selection = "username = \"" + user.getUserName() + "\"";
-
+        try{
```
Format all files with Ctrl+Alt+L from Android Studio. And run code analyzer as well on the files you have updated.

ankitk20 commented on kitkitschoollogger/kitkitlogger/src/main/java/com/maq/kitkitProvider/User.java:
```diff
@@ -25,10 +27,14 @@
     private boolean _unlockWritingBoard;
     private boolean _finishWritingBoardTutorial;
+    private int _gamesClearedInTotal_L;
+    private int _gamesClearedInTotal_M;
```
Please add comments in front of these variables for better understanding

ankitk20 commented on launcher/app/src/main/AndroidManifest.xml:
```diff
@@ -9,51 +9,62 @@
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
     <uses-permission android:name="android.permission.CHANGE_WIFI_STATE" />
-    <uses-permission android:name="android.permission.CHANGE_NETWORK_STATE" />-->
+    <uses-permission android:name="android.permission.CHANGE_NETWORK_STATE" />
```
Take a merge from master and remove commented code wherever applicable

Commit:
Made Image to appear in circle and made multi_user.sml responsive and also made sliding_mages.xml responsive

onetruffle committed:
Resolve cpp issues of severity "style"

onetruffle committed:
Resolve cpp issues

Commit:
Fixed the UI and Edit User

ankitk20 commented on launcher/app/build.gradle:
```diff
@@ -76,6 +76,7 @@ dependencies {
     })
     compile 'com.android.support:appcompat-v7:25.1.0'
+    compile 'de.hdodenhof:circleimageview:3.0.0'
     compile 'com.android.support:design:25.1.0'
```
Use _api_ or _implementation_ instead of _compile_

SanchitTaliyan replied:
I tried but it's not able to find a file required in implementation mode

ankitk20 commented on launcher/app/src/main/AndroidManifest.xml:
```diff
@@ -9,51 +9,62 @@
-    <uses-permission android:name="android.permission.CHANGE_NETWORK_STATE" />-->
+    <uses-permission android:name="android.permission.CHANGE_NETWORK_STATE" />
+    -->
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
```
Check if this is required

SanchitTaliyan replied:
It was there before, for wifi access and network access, might need it in future.

ankitk20 commented on launcher/app/src/main/AndroidManifest.xml:
```diff
@@ -9,51 +9,62 @@
                 <category android:name="android.intent.category.DEFAULT" />
                 <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
+            </intent-filter>-->
```
Remove this commented chunk

SanchitTaliyan replied:
Done

ankitk20 commented on launcher/app/src/main/java/com/maq/xprize/kitkitlauncher/hindi/SlidingPagerAdapter.java:
```diff
@@ -0,0 +1,85 @@
+    }
+
+
+    @Override
```
Avoid leaving empty lines

ankitk20 commented on launcher/app/src/main/java/com/maq/xprize/kitkitlauncher/hindi/SlidingPagerAdapter.java:
```diff
@@ -0,0 +1,85 @@
+
+    @Override
+    public boolean isViewFromObject(View view, Object object) {
+        return view == object;
```
== will always return false. Use _equals()_
https://www.geeksforgeeks.org/difference-equals-method-java/

SanchitTaliyan replied:
I checked online, we should use == only here for equating.

ankitk20 commented on launcher/app/src/main/java/com/maq/xprize/kitkitlauncher/hindi/UserNameActivity.java:
```diff
@@ -82,18 +88,12 @@ public void refreshUI() {
 
         User user = ((LauncherApplication) getApplication()).getDbHandler().getCurrentUser();
-        mTvUserNo.setText(getString(R.string.user_no) + " " + user.getUserName().replace("user", ""));
+        mTvUserNo.setText(getString(R.string.user_no) + " " + user.getUserName().replace("user" , ""));
```
Use Ctrl+Alt+L in all files for autoformatting. Press Ctrl+Shift+Alt+L to see more options

SanchitTaliyan replied:
Done

ankitk20 commented on launcher/app/src/main/java/com/maq/xprize/kitkitlauncher/hindi/UserNameActivity.java:
```diff
@@ -147,6 +147,17 @@ public void onSelectedNumber(int number) {
                     KitkitDBHandler dbHandler = ((LauncherApplication) getApplication()).getDbHandler();
                     ArrayList<User> users = dbHandler.getUserList();
+                    try {
+                        schoolContext = getApplicationContext().createPackageContext("com.maq.xprize.kitkitschool.hindi", 0);
```
Use method to fetch package name instead of hardcoding

ankitk20 commented on launcher/app/src/main/res/layout/activity_multi_user.xml:
```diff
@@ -0,0 +1,163 @@
+        android:layout_marginTop="40dp"
+        android:layout_marginBottom="8dp"
+        android:gravity="center"
+        android:text="Select User\nउपयोगकर्ता चयन"
```
Please use $#$ as delimiter instead of \n everywhere

SanchitTaliyan replied:
\n is used for printing Hindi text in next line

ankitk20 commented on launcher/app/src/main/res/layout/activity_select_user.xml:
```diff
@@ -0,0 +1,207 @@
+        android:background="@drawable/roundtextview"
+        android:fontFamily="Segoe UI"
+        android:gravity="center_horizontal"
+        android:text="Maths\\गणित"
```
Use $#$ instead of \\ everywhere

SanchitTaliyan replied:
Please can you explain me why to use $#$.

ankitk20 commented on launcher/app/src/main/res/layout/activity_setting.xml:
```diff
@@ -496,7 +496,7 @@
                 </FrameLayout>
 
-                <!-- User name -->
+                <!-- User name -->     <!-- here should be my Add Button -->
```
Why is this added?

SanchitTaliyan replied:
It is not my change.

ankitk20 commented on mainapp/Classes/Managers/UserManager.cpp:
```diff
@@ -390,6 +390,16 @@ bool UserManager::isGameCleared(string levelID, int day, int gameIndex)
+        gamesClearedInTotal++;
+        UserDefault::getInstance()->setIntegerForKey(gamesClearedInTotalKey.c_str(), gamesClearedInTotal);
+    }
+    ///////////////////////////////
```
Please comment properly. This code repo is public.

Commit:
Updated the version and obb size

Commit:
Revert "Updated the version and obb size"

This reverts commit e6fca9b0f9f8314a66150a6eb2a4f3ef5636220c.

Commit:
Fixed Minimum user age and Back button UI and English,Maths progress layout.

Commit:
Merge branch 'UserProfile_ProgressTracker_Integration' of https://github.com/maqsoftware/GLEXP-Team-KitkitSchool-Hindi into UserProfile_ProgressTracker_Integration

Commit:
Fixed Select User UI

onetruffle committed:
Merge branch 'SaitarunI/StaticCodeAnalysis_6_14' into UserProfile_ProgressTracker_Integration

Commit:
Resolved Comments

Commit:
Fixed the Back Button and User Image

Commit:
Fixed the usr image and button width in edit user

Commit:
Updated the logic to fetch the user details

Commit:
UI chenges

Commit:
Merge branch 'UserProfile_ProgressTracker_Integration' of https://github.com/maqsoftware/GLEXP-Team-KitkitSchool-Hindi into UserProfile_ProgressTracker_Integration

Commit:
Fixed Current User and Delete User and Add User method

Commit:
Merge branch 'newmaster' into UserProfile_ProgressTracker_Integration

Commit:
Merge branch 'newmaster' into UserProfile_ProgressTracker_Integration

Commit:
Fixed the register button, Back Button and improved UI

Commit:
Merge branch 'UserProfile_ProgressTracker_Integration' of https://github.com/maqsoftware/GLEXP-Team-KitkitSchool-Hindi into UserProfile_ProgressTracker_Integration

Commit:
Merged with master

Commit:
Removed unused methods and fixed the files

Commit:
Updated the files

Commit:
Removed Display Name, Fixed the Launcher first screen, added back button in update UI

Commit:
Added previus and next image button. Also fixed user image in main activity.
